Mathlib/Order/FixedPoints.lean

```diff
@@ -0,0 +1,325 @@
+#align order_hom.lfp_le_fixed OrderHom.lfp_le_fixed
+
+theorem le_lfp {a : α} (h : ∀ b, f b ≤ b → a ≤ b) : a ≤ lfp f :=
+  le_infₛ h
+#align order_hom.le_lfp OrderHom.le_lfp
+
+-- porting note: for the rest of the file, replace the dot notation `_.lfp` with `lfp _`
+-- same for `_.gfp`, `_.dual`
+-- Probably related to https://github.com/leanprover/lean4/issues/1910
+theorem map_le_lfp {a : α} (ha : a ≤ lfp f) : f a ≤ lfp f :=
+  f.le_lfp fun _ hb => (f.mono <| le_infₛ_iff.1 ha _ hb).trans hb
+#align order_hom.map_le_lfp OrderHom.map_le_lfp
+
+@[simp]
+theorem map_lfp : f (lfp f) = lfp f :=
+  have h : f (lfp f) ≤ lfp f := f.map_le_lfp le_rfl
+  h.antisymm <| f.lfp_le <| f.mono h
+#align order_hom.map_lfp OrderHom.map_lfp
+
+theorem isFixedPt_lfp : IsFixedPt f (lfp f) :=
+  f.map_lfp
+#align order_hom.is_fixed_pt_lfp OrderHom.is_fixed_pt_lfp
+
+theorem lfp_le_map {a : α} (ha : lfp f ≤ a) : lfp f ≤ f a :=
+  calc
+    lfp f = f (lfp f) := f.map_lfp.symm
+    _ ≤ f a := f.mono ha
+
+#align order_hom.lfp_le_map OrderHom.lfp_le_map
+
+theorem isLeast_lfp_le : IsLeast { a | f a ≤ a } (lfp f) :=
+  ⟨f.map_lfp.le, fun _ => f.lfp_le⟩
+#align order_hom.is_least_lfp_le OrderHom.is_least_lfp_le
+
+theorem is_least_lfp : IsLeast (fixedPoints f) (lfp f) :=
+  ⟨f.is_fixed_pt_lfp, fun _ => f.lfp_le_fixed⟩
+#align order_hom.is_least_lfp OrderHom.is_least_lfp
+
+-- porting note: replaced `set` with `let` below, unknown tactic
+theorem lfp_induction {p : α → Prop} (step : ∀ a, p a → a ≤ lfp f → p (f a))
+    (hSup : ∀ s, (∀ a ∈ s, p a) → p (supₛ s)) : p (lfp f) := by
+  let s := { a | a ≤ lfp f ∧ p a }
+  specialize hSup s fun a => And.right
+  suffices : supₛ s = lfp f
+  exact this ▸ hSup
+  have h : supₛ s ≤ lfp f := supₛ_le fun b => And.left
+  have hmem : f (supₛ s) ∈ s := ⟨f.map_le_lfp h, step _ hSup h⟩
+  exact h.antisymm (f.lfp_le <| le_supₛ hmem)
+#align order_hom.lfp_induction OrderHom.lfp_induction
+
+theorem le_gfp {a : α} (h : a ≤ f a) : a ≤ gfp f :=
+  le_supₛ h
+#align order_hom.le_gfp OrderHom.le_gfp
+
+theorem gfp_le {a : α} (h : ∀ b, b ≤ f b → b ≤ a) : gfp f ≤ a :=
+  supₛ_le h
+#align order_hom.gfp_le OrderHom.gfp_le
+
+theorem isFixedPt_gfp : IsFixedPt f (gfp f) :=
+  f.dual.is_fixed_pt_lfp
+#align order_hom.is_fixed_pt_gfp OrderHom.is_fixed_pt_gfp
+
+@[simp]
+theorem map_gfp : f (gfp f) = gfp f :=
+  f.dual.map_lfp
+#align order_hom.map_gfp OrderHom.map_gfp
+
+theorem map_le_gfp {a : α} (ha : a ≤ gfp f) : f a ≤ gfp f :=
+  f.dual.lfp_le_map ha
+#align order_hom.map_le_gfp OrderHom.map_le_gfp
+
+theorem gfp_le_map {a : α} (ha : gfp f ≤ a) : gfp f ≤ f a :=
+  f.dual.map_le_lfp ha
+#align order_hom.gfp_le_map OrderHom.gfp_le_map
+
+theorem isGreatest_gfp_le : IsGreatest { a | a ≤ f a } (gfp f) :=
+  f.dual.is_least_lfp_le
+#align order_hom.is_greatest_gfp_le OrderHom.is_greatest_gfp_le
+
+theorem is_greatest_gfp : IsGreatest (fixedPoints f) (gfp f) :=
+  f.dual.is_least_lfp
+#align order_hom.is_greatest_gfp OrderHom.is_greatest_gfp
```

Comment:
```suggestion
theorem isGreatest_gfp_le : IsGreatest { a | a ≤ f a } (gfp f) :=
  f.dual.is_least_lfp_le
#align order_hom.is_greatest_gfp_le OrderHom.isGreatest_gfp_le

theorem isGreatest_gfp : IsGreatest (fixedPoints f) (gfp f) :=
  f.dual.is_least_lfp
#align order_hom.is_greatest_gfp OrderHom.isGreatest_gfp
```
Sorry, I forgot to say that you need to fix the aligns as well (and there will most likely be errors down the line).